schemas/workflows/workflow-step.json

```diff
@@ -72,20 +72,26 @@
   },
   "anyOf": [
     {
-      "not": {
-        "anyOf" :[
-          {
-            "required": ["on_failure"]
-          },
-          {
-            "required": ["on_success"]
-          }
-        ]
-      }
+      "anyOf" :[
+        {
+          "required": ["on_failure"]
+        },
+        {
+          "required": ["on_success"]
+        }
+      ]
     },
     {
-      "not": {
-        "required": ["on_completion"]
+      "required": ["on_completion"]
+    },
+    {
+      "properties": {
+        "type" : {
+          "type" : "string",
+          "enum": [
+            "end"
```

Comment:
What's that?

Reply:
This is to remove the need for the end step to have on_completion or on_failure/on_success.